include/fmt/chrono.h

```diff
@@ -428,14 +432,20 @@ template <typename Char> struct formatter<std::tm, Char> {
   }
 
   template <typename FormatContext>
-  auto format(const std::tm& tm, FormatContext& ctx) const
+  auto format(const std::tm& tm, FormatContext& ctx, int subseconds) const
       -> decltype(ctx.out()) {
     basic_memory_buffer<Char> tm_format;
     tm_format.append(specs.begin(), specs.end());
     // By appending an extra space we can distinguish an empty result that
     // indicates insufficient buffer size from a guaranteed non-empty result
     // https://github.com/fmtlib/fmt/issues/2238
-    tm_format.push_back(' ');
+    auto hasS = std::find(tm_format.begin(), tm_format.end(), 'S') != tm_format.end();
+    auto hasT = std::find(tm_format.begin(), tm_format.end(), 'T') != tm_format.end();
+    auto writeSubseconds = (hasS || hasT) && (subseconds != 0);
+    if (writeSubseconds)
+      tm_format.push_back('.');
```

Comment:
Does this need to check locale an insert the preferred decimal separator?

Reply:
It should check, but I don't know how to get the locale in the chrono formatter. I'll try to grep for anything related to locale.

Reply:
Locale can be addressed separately, the default is locale-independent anyway.